source/World.cpp

```diff
@@ -288,22 +288,397 @@ namespace GTA
-
-		for (int i = 1; i < 40; i++)
-		{
-			Native::Function::Call(Native::Hash::GET_NTH_CLOSEST_VEHICLE_NODE, position.X, position.Y, position.Z, i, outPos, 1, 0x40400000, 0);
-			const Math::Vector3 newPos = outPos->GetResult<Math::Vector3>();
-
-			if (!Native::Function::Call<bool>(Native::Hash::IS_POINT_OBSCURED_BY_A_MISSION_ENTITY, newPos.X, newPos.Y, newPos.Z, 5.0f, 5.0f, 5.0f, 0))
-			{
-				return newPos;
-			}
-		}
-		return Math::Vector3();
-	}
+	Math::Vector3 World::GetNextPositionOnSidewalk(Math::Vector2 position)
+	{
+		return World::GetNextPositionOnSidewalk(Math::Vector3(position.X, position.Y, 0));
+	}
+	Math::Vector3 World::GetNextPositionOnSidewalk(Math::Vector3 position)
+	{
+		Native::OutputArgument ^outPos = gcnew Native::OutputArgument();
+
+		if (Native::Function::Call<bool>(Native::Hash::GET_SAFE_COORD_FOR_PED, position.X, position.Y, position.Z, true, outPos, 0))
+		{
+			return outPos->GetResult<Math::Vector3>();
+		}
+		else if (Native::Function::Call<bool>(Native::Hash::GET_SAFE_COORD_FOR_PED, position.X, position.Y, position.Z, false, outPos, 0))
+		{
+			return outPos->GetResult<Math::Vector3>();
+		}
+		else return Math::Vector3::Zero;
+	}
+	Math::Vector3 World::GetNextPositionOnStreet(Math::Vector3 position)
+	{
+		return World::GetNextPositionOnStreet(position, false);
+	}
+	Math::Vector3 World::GetNextPositionOnStreet(Math::Vector2 position, bool unoccupied)
+	{
+		return World::GetNextPositionOnStreet(Math::Vector3(position.X, position.Y, 0), unoccupied);
+	}
+	Math::Vector3 World::GetNextPositionOnStreet(Math::Vector3 position, bool unoccupied)
+	{
+		Native::OutputArgument ^outPos = gcnew Native::OutputArgument();
+
+		if (unoccupied)
+		{
+			for (int i = 1; i < 40; i++)
+			{
+				Native::Function::Call(Native::Hash::GET_NTH_CLOSEST_VEHICLE_NODE, position.X, position.Y, position.Z, i, outPos, 1, 0x40400000, 0);
+				const Math::Vector3 newPos = outPos->GetResult<Math::Vector3>();
+
+				if (!Native::Function::Call<bool>(Native::Hash::IS_POINT_OBSCURED_BY_A_MISSION_ENTITY, newPos.X, newPos.Y, newPos.Z, 5.0f, 5.0f, 5.0f, 0))
+				{
+					return newPos;
+				}
+			}
+		}
+		else if (Native::Function::Call<bool>(Native::Hash::GET_NTH_CLOSEST_VEHICLE_NODE, position.X, position.Y, position.Z, 1, outPos, 1, 0x40400000, 0))
+		{
+			return outPos->GetResult<Math::Vector3>();
+		}
+		return Math::Vector3();
+	}
+	Math::Vector3 World::GetSafeCoordForPed(Math::Vector3 position)
+	{
+		return World::GetSafeCoordForPed(position.X, position.Y, position.Z, true, 0);
+	}
+	Math::Vector3 World::GetSafeCoordForPed(Math::Vector3 position, bool sidewalk)
+	{
+		return World::GetSafeCoordForPed(position.X, position.Y, position.Z, sidewalk, 0);
+	}
+	Math::Vector3 World::GetSafeCoordForPed(Math::Vector3 position, bool sidewalk, int flags)
+	{
+		return World::GetSafeCoordForPed(position.X, position.Y, position.Z, sidewalk, flags);
+	}
+	Math::Vector3 World::GetSafeCoordForPed(float x, float y, float z, bool sidewalk, int flags)

```

Comment:
This function should be removed and instead use the one above it. Its bad practices to have functions for positions that take seperate float arguments